### PR TITLE
Implement a `sort` operator

### PR DESCRIPTION
**Pull request by dominiklohmann**
This is a very basic implementation of a sort operator that comes with a few fundamental limitations:
- All data is kept in memory
- There is no windowing support
- Extension types are not supported for sorting (ip, subnet, enum).
- Sorting only works by one column at a time
- The resulting slices of the operator have a size of 1, which leads to very slow data access in following operators

But other than that it works as you'd expect.

```[tasklist]
### Tasks
- [x] Implement in-memory sorting
- [x] Write follow-up stories for the above-mentioned limitations
- [x] Write tests
- [x] Write documentation (including known limitations)
- [x] Add a changelog entry
```

### CODE REVIEW

**Comment by mavam** [comment]
@dominiklohmann I pushed a commit with an updated documentation.

File: web/docs/understand/operators/transformations/sort.md
Comment (dominiklohmann) [inline web/docs/understand/operators/transformations/sort.md:8]: I changed this to `sort <field> [<asc|desc>] [<nulls-first>|<nulls-last>]` since we talked this morning; I think we should agree on the exact syntax we want before merging this. What we have implemented now is my current preference.

Comment (mavam) [reply web/docs/understand/operators/transformations/sort.md]: Okay, I've updated the docs again.

It looks like you haven't pushed your latest changes, though. Feel free to do so now.

**Comment by mavam** [comment]
One thing I noticed during testing: just `sort` by itself doesn't work, even if the input has just a single field. I would argue we note this down and tackle it during the other follow-ups.

**Comment by mavam** [comment]
Added another commit to test `nulls-first`. (Currently failing)

**Comment by dominiklohmann** [comment]
> One thing I noticed during testing: just `sort` by itself doesn't work, even if the input has just a single field. I would argue we note this down and tackle it during the other follow-ups.

The sensible default if the field is omitted is likely `sort . asc nulls-last`, but we can't express that in the expression language yet. I've added a story for this.

**Review by mavam — APPROVED** [review]
Overall, this works as expected. I added tests and they now work.

Unfortunately I lack the Arrow depth to review what the code does in detail. If you need a second pair of eyes from @tenzir/pipelines-squad, I'd ask there. Otherwise my approval is only based on high-level skimming and testing.

File: libvast/builtins/operators/sort.cpp
Comment (dominiklohmann) [reply libvast/builtins/operators/sort.cpp:116]: Yes, that's still the case in Arrow 12, which I'm using locally. I have a list of things we can swap out with the recent improvements for extension types, but this isn't one of them, sadly.

File: libvast/include/vast/type.hpp
Comment (dominiklohmann) [reply libvast/include/vast/type.hpp]: I made this change in b6b11087611d1f1a30e6b24b51df3ce7aac39308

**Pull request merged**
